frontend/src/components/Data/DatasetTable.js

```diff
@@ -4,14 +4,24 @@ import {observer} from "mobx-react";
 import PropTypes from "prop-types";
 
 import {columns, columnHeaders} from "../../constants/dataConstants";
+import {Dtype} from "../../constants/dataConstants";
 
 @observer
 class DatasetTable extends Component {
     render() {
         const {dataset} = this.props,
             columnNames = columns[dataset.dtype],
-            width = `${100 / columnNames.length}%`;
-
+            width = `${100 / columnNames.length}%`,
+            dose_group = _.uniq(dataset.doses),
+            getResponseList = function(key) {
```

Comment:
can you use a group by? and more lodash patterns, they're easier to understand

```javascript
data=[{dose:1, r:1},{dose:1, r:1},{dose:1, r:1},{dose:2, r:1},{dose:2, r:1}]
grouped= _.groupBy(data, d=>d.dose)
_.values(grouped)[0].map(d=>d.r.toString()).join(", ") 
```

you can just pull up on a console on https://lodash.com/  to play around with these commands